query_optimizer/expressions/InValueList.hpp

```diff
@@ -0,0 +1,158 @@
+  }
+
+  ExpressionPtr copyWithNewChildren(
+      const std::vector<ExpressionPtr> &new_children) const override {
+    DCHECK_EQ(new_children.size(), children().size());
+    std::vector<ScalarPtr> new_match_expressions;
+    for (std::size_t idx = 1; idx < new_children.size(); ++idx) {
+      new_match_expressions.emplace_back(
+          std::static_pointer_cast<const Scalar>(new_children[idx]));
+    }
+    return Create(std::static_pointer_cast<const Scalar>(new_children[0]),
+                  new_match_expressions);
+  }
+
+  std::vector<AttributeReferencePtr> getReferencedAttributes() const override {
+    std::vector<AttributeReferencePtr> referenced_attrs =
+        test_expression_->getReferencedAttributes();
+    for (const ScalarPtr &match_expression : match_expressions_) {
+      const std::vector<AttributeReferencePtr> referenced_attrs_in_match_expr =
+          match_expression->getReferencedAttributes();
+      referenced_attrs.insert(referenced_attrs.end(),
+                              referenced_attrs_in_match_expr.begin(),
+                              referenced_attrs_in_match_expr.end());
+    }
+    return referenced_attrs;
+  }
+
+  ::quickstep::Predicate* concretize(
+      const std::unordered_map<ExprId, const CatalogAttribute*> &substitution_map) const override;
+
+  /**
+   * @brief Create an IN predicate with a value list.
+   *
+   * @param test_expression The expression to test with a value list.
+   * @param match_expressions Expressions to search for the value of the test_expression.
+   *
+   * @return An immutable IN predicate node with a value list.
+   */
+  static InValueListPtr Create(const ScalarPtr &test_expression,
+                               const std::vector<ScalarPtr> &match_expressions) {
+    return InValueListPtr(new InValueList(test_expression, match_expressions));
+  }
+
+ protected:
+  void getFieldStringItems(
+      std::vector<std::string> *inline_field_names,
+      std::vector<std::string> *inline_field_values,
+      std::vector<std::string> *non_container_child_field_names,
+      std::vector<OptimizerTreeBaseNodePtr> *non_container_child_fields,
+      std::vector<std::string> *container_child_field_names,
+      std::vector<std::vector<OptimizerTreeBaseNodePtr>> *container_child_fields) const override;
+
+ private:
+  InValueList(const ScalarPtr &test_expression,
+              const std::vector<ScalarPtr> &match_expressions)
+      : test_expression_(test_expression),
+        match_expressions_(match_expressions) {
+    addChild(test_expression_);
+    for (const ScalarPtr &match_expression : match_expressions_) {
+      addChild(match_expression);
+    }
+  }
+
+  ScalarPtr test_expression_;
+  std::vector<ScalarPtr> match_expressions_;
+
+  DISALLOW_COPY_AND_ASSIGN(InValueList);
+};
+
+/** @} */
+
+}  // namespace expressions
+}  // namespace optimizer
+}  // namespace quickstep
+
+
+#endif /* QUICKSTEP_QUERY_OPTIMIZER_EXPRESSIONS_IN_VALUE_LIST_HPP_ */
```

Comment:
Please remove a unnecessary empty line above. Also, ditto for `//` instead of `/**/`. Thanks!